pyclam/manifold.py

```diff
@@ -965,8 +965,31 @@ def build_tree(self, *criterion) -> 'Manifold':
                 self.layers.append(Graph(*clusters))
             else:
                 break
+        self.cache.clear()
+        return self
+
+    def add_graphs(self, *criterion) -> 'Manifold':
+        """ Uses the given selection criteria to add more graphs to the Manifold. """
+        from pyclam.criterion import SelectionCriterion
+
+        if not all((isinstance(c, SelectionCriterion) for c in criterion)):
+            raise ValueError(f'Only Selection Criteria are allowed for building graphs.')
+
+        logging.info(f'building graphs with {len(criterion)} Selection Criteria.')
+        self.graphs.extend([Graph(*select(self.root)).build_edges() for select in criterion])
+        self.cache.clear()
```

Comment:
I'm currently against clearing caches by default this way. What's the argument for forcibly clearing them at the end? I'd almost rather see it at the front, or just selectively clearing what it needs to work.